### PR TITLE
[forward-port] More upload test changes for clickhouse (#42531)

### PR DESCRIPTION
**Pull request by calherries**
This is a forward-port of https://github.com/metabase/metabase/pull/42531

This was developed on 49 originally because the clickhouse driver only works with 49 currently, and I wanted to check it worked correctly.

Compared to the original PR I didn't add anything new, although there were many merge conflicts.

### CODE REVIEW

File: test/metabase/upload_test.clj [hunk 1]
```diff
@@ -1262,18 +1262,46 @@
...
+                                              (ordered-map/ordered-map :offset_datetime offset-dt-type))
+                           :rows []})]
+                  (let [csv-rows ["offset_datetime"
+                                  "2020-02-02T02:02:02+02:00"]
+                        file  (csv-file-with csv-rows (mt/random-name))]
+                    (is (some? (update-csv! action {:file file, :table-id (:id table)})))
+                    (testing "Check the data was uploaded into the table correctly"
+                      (is (= (set (updated-contents
+                                   action
+                                   ;; TODO
```
Comment (crisptrutski) [inline test/metabase/upload_test.clj:1300]: Can you add more color to this comment and perhaps an issue link?

Comment (crisptrutski) [reply test/metabase/upload_test.clj:1300]: Oops nevermind, see I jumped the gun on reviewing.

Comment (crisptrutski) [reply test/metabase/upload_test.clj:1300]: Oops again, see you did request a review and I tripped over the way GH implicitly unassigns the team after I comment.

**Review by crisptrutski — APPROVED** [review]
LGTM besides nit

File: test/metabase/upload_test.clj (hunk 1 above)
Comment (calherries) [reply test/metabase/upload_test.clj]: Oops, that wasn't meant to be left there. Removed

**Pull request merged**